### ISSUE TITLE
Unable to create UDTs greater than 128 bytes on Linux and OS X

### ISSUE THREAD

**Issue by eriknw** [DrTimothyAldenDavis/GraphBLAS#136]
I am unable to create user-defined types greater than 128 bytes on OS X and Linux.  I don't know about Windows.

I am using compiled GraphBLAS binaries from conda-forge.  Linux uses GCC version 10.  You can see compiler versions here:
https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml

I see the size of UDTs is limited to `GB_VLA_MAXSIZE` (which is 128) in `Source/GxB_Type_new.c` if `! GB_HAS_VLA`.  I would expect to _not_ have this limitation on Linux or OS X, because I see `GxB_STDC_VERSION` == `201710` in our cffi binding.

Any idea why this might be?  Are there additional build flags we need to set to enable this?  How should I investigate further?  Do you test this?

**Comment by DrTimothyAldenDavis** [comment]
That's odd.  The user-defined type sizes should only be limited on Windows, not Linux or gcc.  Can you call this and see what it reports?

```
char *compiler ;
int compiler_version [3] ;
GxB_Global_Option_get (GxB_COMPILER_NAME, &compiler) ;
GxB_Global_Option_get (GxB_COMPILER_VERSION, compiler_version) ;
printf ("GraphBLAS compiled with %s version %d.%d.%d\n", compiler,
  compiler_version [0], compiler_version [1], compiler_version [2]) ;
```

For Windows, if 128 bytes is too small, I could make that bigger.  It just needs to be a fixed number of bytes, because of the limitations of the Windows MS Visual Studio compiler.

**Comment by DrTimothyAldenDavis** [comment]
Even on Windows, using gcc should allow user-defined types of arbitrary size.  See https://github.com/DrTimothyAldenDavis/GraphBLAS/blob/stable/Source/GB_compiler.h

**Comment by eriknw** [comment]
Yeah, it's weird.  I get `"GNU gcc 10.3.0"` and `[10, 3, 0]` for `GxB_COMPILER_NAME` and `GxB_COMPILER_VERSION`.

**Comment by DrTimothyAldenDavis** [comment]
That is very strange.  This line in GB_compiler.h should trigger:
https://github.com/DrTimothyAldenDavis/GraphBLAS/blob/4bb44bb93a30b54c7039d1c6f5d87e3c85ec45a5/Source/GB_compiler.h#L219

And this body of code should be disabled:
https://github.com/DrTimothyAldenDavis/GraphBLAS/blob/4bb44bb93a30b54c7039d1c6f5d87e3c85ec45a5/Source/GxB_Type_new.c#L46 

Can you try adding a printf inside that body of code in GxB_Type_new, where I check the type size with GB_VLA_MAXSIZE (which is equal to 128)?  That should not be triggered.

**Comment by eriknw** [comment]
Done.  The print statement inside `#if ( ! GB_HAS_VLA ) ` block triggers for me on Linux.  Still using GCC.

Also, we also see this problem in the Python wheels we build for `python-suitesparse-graphblas`, so it's not limited to conda-forge.

**Comment by eriknw** [comment]
I appear to be hitting this block:
https://github.com/DrTimothyAldenDavis/GraphBLAS/blob/4bb44bb93a30b54c7039d1c6f5d87e3c85ec45a5/Source/GB_compiler.h#L221-L226
which, uh, makes no sense, b/c I also see `GxB_STDC_VERSION` as `201112` (or `201710`), which is indeed greater than `199901` (the previous elif condition). 😕

**Comment by DrTimothyAldenDavis** [comment]
Oops.  I see the problem.  GB_compiler.h is #included in GB.h before #include'ing GraphBLAS.h, and GxB_STDC_VERSION is not yet defined.  So GB_compiler.h triggers line 224 for all compilers.

**Comment by DrTimothyAldenDavis** [comment]
See my most recent push to the master branch, a one-line change to Source/GB_compiler.h.

**Comment by DrTimothyAldenDavis** [comment]
Fixed in v7.0.4.
